File: English/Leads/Python/GPT/app/models/lead.py
from datetime import date, datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel, EmailStr, Field


class LeadStatus(str, Enum):
    new = "new"
    contacted = "contacted"
    qualified = "qualified"
    lost = "lost"


class LeadBase(BaseModel):
    full_name: str = Field(min_length=2, max_length=160)
    email: EmailStr
    phone: Optional[str] = Field(default=None, max_length=40)
    company: Optional[str] = Field(default=None, max_length=160)
    position: Optional[str] = Field(default=None, max_length=160)
    source: str = Field(min_length=2, max_length=80)
    status: LeadStatus = LeadStatus.new
    capture_date: date


class LeadCreate(LeadBase):
    pass


class LeadUpdate(BaseModel):
    full_name: Optional[str] = Field(default=None, min_length=2, max_length=160)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(default=None, max_length=40)
    company: Optional[str] = Field(default=None, max_length=160)
    position: Optional[str] = Field(default=None, max_length=160)
    source: Optional[str] = Field(default=None, min_length=2, max_length=80)
    status: Optional[LeadStatus] = None
    capture_date: Optional[date] = None


class LeadRead(LeadBase):
    id: str
    created_at: datetime
    updated_at: datetime
    interaction_count: int = 0


class LeadListResponse(BaseModel):
    items: list[LeadRead]
    total: int
    page: int
    page_size: int
    total_pages: int


class InteractionCreate(BaseModel):
    interaction_type: str = Field(min_length=2, max_length=60)
    summary: str = Field(min_length=2, max_length=160)
    notes: Optional[str] = Field(default=None, max_length=4000)
    happened_at: datetime


class InteractionRead(InteractionCreate):
    id: str
    created_at: datetime


class LeadDetailResponse(BaseModel):
    lead: LeadRead
    interactions: list[InteractionRead]

File: English/Leads/Go/Claude/internal/models/lead.go
package models

import "time"

type Lead struct {
	ID          string    `json:"id" firestore:"-"`
	FullName    string    `json:"full_name" firestore:"full_name"`
	Email       string    `json:"email" firestore:"email"`
	Phone       string    `json:"phone" firestore:"phone"`
	Company     string    `json:"company" firestore:"company"`
	Position    string    `json:"position" firestore:"position"`
	Source      string    `json:"source" firestore:"source"`
	Status      string    `json:"status" firestore:"status"`
	CaptureDate time.Time `json:"capture_date" firestore:"capture_date"`
	Notes       string    `json:"notes" firestore:"notes"`
	UserID      string    `json:"user_id" firestore:"user_id"`
	CreatedAt   time.Time `json:"created_at" firestore:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" firestore:"updated_at"`
}

type Interaction struct {
	ID        string    `json:"id" firestore:"-"`
	LeadID    string    `json:"lead_id" firestore:"lead_id"`
	UserID    string    `json:"user_id" firestore:"user_id"`
	Type      string    `json:"type" firestore:"type"`
	Notes     string    `json:"notes" firestore:"notes"`
	CreatedAt time.Time `json:"created_at" firestore:"created_at"`
}

type LeadFilter struct {
	Status      string `json:"status"`
	Source      string `json:"source"`
	DateFrom    string `json:"date_from"`
	DateTo      string `json:"date_to"`
	Search      string `json:"search"`
	Page        int    `json:"page"`
	PageSize    int    `json:"page_size"`
	SortBy      string `json:"sort_by"`
	SortOrder   string `json:"sort_order"`
}

type PaginatedResponse struct {
	Data       interface{} `json:"data"`
	Total      int         `json:"total"`
	Page       int         `json:"page"`
	PageSize   int         `json:"page_size"`
	TotalPages int         `json:"total_pages"`
}

var ValidStatuses = []string{"new", "contacted", "qualified", "lost"}
var ValidSources = []string{"referral", "website", "event", "social_media", "cold_call", "other"}
